src/Hyperion/Run.hs

```diff
@@ -81,6 +89,20 @@ runBenchmark samplef bk0 =
       put xs
       return x
 
+-- | Run a benchmark on varying sample sizes
+runBenchmarkVaryingSizes :: [(Batch () -> IO Sample)] -> Benchmark -> IO (HashMap Text [Sample])
+runBenchmarkVaryingSizes samplef bk0 = formatSamples $ sequence (map (\samplef' -> runBenchmark samplef' bk0) samplef)
+
+-- | Set of functions to massage the data into the appropriate type
+formatSamples :: IO([HashMap Text Sample]) -> IO (HashMap Text [Sample])
+formatSamples samples = concatenateSamples $ expandSamples samples
+
+expandSamples :: IO ([HashMap Text Sample]) -> IO ([HashMap Text [Sample]])
+expandSamples samples = fmap (map (fmap (\x -> [x]))) samples
+
+concatenateSamples :: IO ([HashMap Text [Sample]]) -> IO (HashMap Text [Sample])
+concatenateSamples samples = fmap (foldl' (HashMap.unionWith(++)) HashMap.empty) samples
+
```

Comment:
Same comment as above applies here:
```
concatenateSamples :: [HashMap Text [Sample]] -> HashMap Text [Sample]
concatenateSamples = foldl' (HashMap.unionWith(++)) HashMap.empty)
```

Then use it as (assuming previous comment is addressed) `fmap (concatenateSamples  . expandSamples) samples`